Automattic/jetpack
# Protect: WAF Phase 2

Project branch for adding the second iteration of the WAF integration in Jetpack Protect.

This PR affects three projects:
* **packages/waf:** Adds separation between the toggling of automatic and manual rules.
* **plugins/jetpack:** Updates firewall settings UI to add the new automatic rule setting toggle.
* **plugins/protect:** Updates the firewall settings UI to add the new automatic rule setting toggle.

#### Changes proposed in this Pull Request:

| Task | PR | Status |
| ---- | -- | ------ |
| Add support for enabling automatic and manual rules independently of each other | #27724 | 🟣 Merged |
| Add a dedicated automatic-rules.php file | #28027 | 🟣 Merged |
| Toggle automatic rules in the Jetpack Settings UI | #27785 | 🟣 Merged |
| Toggle automatic rules in the Protect UI | #27663 | 🟣 Merged |
| Add a dedicated screen for editing manual rules | #27786 | 🟣 Merged |
| Add Waf_Rules_Manager class | #28084 | 🟣 Merged |
| Extract definition of constants from Waf_Runner to Waf_Constants | #28104 | 🟣 Merged |
| Update header to consider firewall off when no rules are turned on | #28326 | 🟣 Merged |
Add rules versioning | D95826-code | 🟣  Merged
Add post-upgrade messaging | #27787 | 🟣  Merged
Add Waf_Stats class | #27970 | 🟣  Merged
Add stats to controls | #28017 | 🟣  Merged
Extract REST API logic from main plugin class | #28217 | 🟣  Merged

#### Other information:


#### Jetpack product discussion



peb6dq-4G-p2

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






* Review all merged PRs, proof read file changes.
* Validate that all tests are passing.
* Smoke test the Jetpack and Protect plugins to ensure the firewall settings work as described.

Bonus Points:
* Install and activate Jetpack Protect. Validate that the "waf" module is automatically enabled.
* Turn on automatic rules. Validate that a `?test=<script>` request is blocked. Validate the contents of the rule files in `wp-content/jetpack-waf/rules/`.
* Turn off automatic rules, and add manual rules. Validate the manual rules apply, but a `?test=<script>` request is not blocked. Validate the contents of the rule files in `wp-content/jetpack-waf/rules/`.
